wdl-lint/src/v1/missing_runtime.rs

```diff
@@ -0,0 +1,76 @@
+//! A lint rule for missing runtime sections.
+
+use wdl_ast::experimental::v1::TaskDefinition;
+use wdl_ast::experimental::v1::Visitor;
+use wdl_ast::experimental::AstToken;
+use wdl_ast::experimental::Diagnostic;
+use wdl_ast::experimental::Diagnostics;
+use wdl_ast::experimental::Span;
+use wdl_ast::experimental::VisitReason;
+
+use super::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the missing runtime rule.
+const ID: &str = "missing_runtime";
+
+/// Creates a "missing runtime section" diagnostic.
+fn missing_runtime_section(task: &str, span: Span) -> Diagnostic {
+    Diagnostic::warning(format!("task `{task}` is missing a runtime section"))
+        .with_code(ID)
+        .with_label("this task is missing a runtime section", span)
+        .with_fix(
+            "add a runtime section to the task with desired cpu, memory and container/docker \
```

Comment:
```suggestion
            "add a runtime section to the task with desired cpu, memory and container \
```
The spec used to only use docker images (hence the validity of the `docker` key), but it's been expanded to (I think, going off the top) any OCI registry. We should remain implementation agnostic here and just say `container`.

Reply:
Another thought. We have another lint planned for "you should really define these runtime keys" and I don't remember what all they are off the top of my head. But this language should include all those keys... Or no keys 🤔 But mentioning `cpu` and not `maxRetries` strikes me as inconsistent (once that rule is implemented). I'd lean towards not mentioning any specific keys in this message, as mentioning everything we will require might be too verbose.

Reply:
I'm also fine with `add a runtime section to the task` with perhaps an additional note of `for a list of mandatory runtime attributes, see https://github.com/openwdl/wdl/blob/wdl-1.1/SPEC.md#mandatory-runtime-attributes`?

Reply:
Although, that part of the spec is more geared towards what an execution engine _must_ support rather than what is required of WDL authors, it seems.

Perhaps just `add a runtime section to the task` and let the other lint rule validate the expected required runtime keys for WDL authors.

Reply:
A little tangent: "Mandatory" has a bit of a dubious meaning in that document, because they aren't actually mandatory to specify. They are mandatory to have a default, but that's an implementation detail for engines that users don't need to be concerned with.
Me and Andrew have complained previously that the WDL spec doc is one of the best sources for new WDL users/writers to learn. And that being the best is not a compliment 😅 There should be a better place to link to for these things, instead of reading through a lengthy SPEC doc. So I'd rather not link there in these contexts, but IDK where _to_ link instead.

For now, maybe just leave it at `add a runtime section to the task` and we can revisit later? IDK I'm torn on the right approach here.

Reply:
I opted for `add a runtime section to the task` and not mention any keys to delegate that to the upcoming rule that will ensure the expected keys are present.

Reply:
O we wrote comments at the same time lol. Looks like we came to the same conclusion though. Sounds good! 👍